Microsoft.NET.Build.Containers/ParseContainerProperties.cs

```diff
@@ -106,11 +101,29 @@ public override bool Execute()
             return !Log.HasLoggedErrors;
         }
 
+        try
+        {
+            if (!ContainerHelpers.NormalizeImageName(ContainerImageName, out string? normalizedImageName))
+            {
+                Log.LogWarning($"{nameof(ContainerImageName)} was not a valid container image name, it was normalized to {normalizedImageName}");
+                NewContainerImageName = normalizedImageName;
+            }
+            else
+            {
+                // name was valid already
+                NewContainerImageName = ContainerImageName;
+            }
+        }
+        catch (ArgumentException)
+        {
+            Log.LogError($"Invalid {nameof(ContainerImageName)}: {{0}}", ContainerImageName);
```

Comment:
This was the final straw, I finally filed https://github.com/dotnet/msbuild/issues/7875.